### ISSUE TITLE
Click event on edge labels

### ISSUE THREAD

**Issue by kandangit** [almende/vis#3261]
Hello,
Want to add a click event for the edges
Problem is, need to keep that in the edit mode the edges can still be draggable (click in this case is to select them)
Is it possible to add a click event / function for the edge labels? That way could edit the edges to drag them and click on the label if want to edit it or other edge properties. is possible or there is another recommended way of doing this?

**Comment by wimrijnders** [comment]
Duplicate of #824.

I must say your timing is remarkable, I was on the verge of closing that issue.

As for other ways, it's possible to use a key combination with the mouse (e.g. control-click) to discern between different actions. Is this of use to you? In that case,  I'll see if I can make some example code.

**Comment by kandangit** [comment]
Hi @wimrijnders ,
many thanks,
yes. I guess for now should be fine. To have a click event for the edge (used for dragging) and a ctrl+click to use for editing edge labels (or nodes rather than dragging them). is that possible?

**Comment by wimrijnders** [comment]
I think it's possible, yes. The basis for control-click is the following:

```js
network.on("click", function(params) {
  if (params.event.srcEvent.ctrlKey === true) {
    var firstNode = params.nodes[0];
    var firstEdge = params.edges[0];
    var msg = 'control click! ';

    if (firstNode !== undefined) {
      alert(msg + 'node id: ' + firstNode);
    } else if (firstEdge !== undefined) {
      alert(msg + 'edge id: ' + firstEdge);
    } else {
      // background click, ignore
    }
  }
});
```

the regular click just falls through. You might want to combine this with the dialogs in the `manipulation` example.

**Comment by kandangit** [comment]
Thanks for the example, worked fine

Could be useful to either allow an event to click on the edge labels.

Or an option to be able to have two buttons after selecting an edge: one for dragging and another for editing it (label, etc). In my case, the edit button always opens a dialog (nodes, edges) so to be consistent wanted to keep it for edges. but in this case lose the dragging functionality as there is not "editDragEdge" similar to existing "editEdge"

For now , I think can have a ctrl+click event on the edges to edit them without using the existing edit button.

thank you

**Comment by kandangit** [comment]
Another option is to have the dialog linked to the "click" event
And just allow the dragging on selecting the edge and using the editEdge function

Is it possible to hide the "editNode" button? From docs, only takes a function (not boolean to disabled it) which is different from the other functions (addEdge, addNode, etc) ?

**Comment by wimrijnders** [comment]
> Could be useful to either allow an event to click on the edge labels.

Yeah. I'm keeping it in the pipeline under issue #824 (implying that I can close this issue when we're done). I'm hereby committed to implementing that.

Your notes on button handling and dialog linked can be implemented already, with clever use of the click handler code example above. But I don't want to go into it in detail, since this issue is solved.

> Is it possible to hide the "editNode" button? From docs, only takes a function (not boolean to disabled it) which is different from the other functions (addEdge, addNode, etc) ?

That's a good observation; it is indeed different from the other functions. I'll have a good look at it to see if I can make the handling more uniform.

In any case, you can hide it by setting `manipulation.editNode: undefined` in the options.

> ...would be good to have editDragMode()...

Isn't this the difference between examples `network/manipulation` and `network/manipulationEditEdgeNodDrag`? So the distinction can be made already.

Perhaps you mean that it should be possible to easily switch between the two modes in the same application? That should be possible as well by combining the two demo's, without changing `vis.js`.

**Comment by kandangit** [comment]
Thanks for the explanations,

sorry the confussion: For dragging edges, thought about a different button. 
When an edge is selected display two buttons "editEdge" to edit the properties and "drag edge" to move it. At the moment I will go for the decission of showing dialog in editEdge function and a ctrl+click event to enable dragging by calling network.manipulation.editMode()


Find it easier to use the editEdge function for this case, as it receives a callback function as a parameter, otherwise have to write the logic which is done by this callback function in a the event function (network.on("click". function(params) {})

**Comment by wimrijnders** [comment]
OK noted. I'm adding this a a low-priority item on my TODO.

**Comment by wimrijnders** [comment]
FYI I picked up on the clicking for labels, see #3410.
